src/OrchardCore.Modules/OrchardCore.DynamicCache/TagHelpers/CacheDependencyTagHelper.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using OrchardCore.Environment.Cache;
+
+namespace OrchardCore.DynamicCache.TagHelpers
+{
+    [HtmlTargetElement("cache-dependency", Attributes = DependencyAttributeName)]
+    public class CacheDependencyTagHelper : TagHelper
+    {
+        private const string DependencyAttributeName = "dependency";
+
+        /// <summary>
+        /// Gets or sets a <see cref="string" /> with the dependency to invalidate the cache with.
+        /// </summary>
+        [HtmlAttributeName(DependencyAttributeName)]
+        public string Dependency { get; set; }
```

Comment:
Why not `Dependencies` allowing to add one or multiple dependencies
By using a `char[] SplitChars = new[] { ',', ' ' }` as in `DynamicCacheTagHelper`

Reply:
@jtkech yes, I thought about that as well, but didn't because that was not what the liquid filter was doing.
And the usage is likely to be similar, as part of a loop, (it was for me anyway) e.g.
```
<cache cache-id="foo">
   var query = myquery;
   foreach(var item in query)
  {
      <cache-dependency dependency="contentitemid:@item.ContentItemId"/>
  }
<cache>
```

But if you think it should be, then I am happy to change it?

Also wondered about making it a selfclosing tag, as it would never have content, but again didn't because it would force it to be used self closing, where many of our other ones like `script` are not forced self closing, even though they don't render content

Reply:
@deanmarcussen okay cool

I just wanted to make sure we didn't miss anything